driver-core/src/main/java/com/datastax/driver/core/QueryOptions.java

```diff
@@ -299,7 +301,19 @@ public QueryOptions setMetadataEnabled(boolean enabled) {
         boolean wasEnabled = this.metadataEnabled;
         this.metadataEnabled = enabled;
         if (!wasEnabled && enabled && manager != null) {
-            manager.submitSchemaRefresh(null, null, null, null); // will also refresh token map
+            // This is roughly the same as what we do in ControlConnection.tryConnect():
+            // 1. call submitNodeListRefresh() first to
+            // be able to compute the token map for the first time,
+            // which will be incomplete due to the lack of keyspace metadata
+            Futures.addCallback(manager.submitNodeListRefresh(), new MoreFutures.SuccessCallback<Void>() {
+                @Override
+                public void onSuccess(Void result) {
+                    // 2. then call submitSchemaRefresh() to
+                    // refresh schema metadata and re-compute the token map
+                    // this time with information about keyspaces
+                    manager.submitSchemaRefresh(null, null, null, null);
+                }
+            });
```

Comment:
Do we need to refresh the node list here? `metadataEnabled == false` doesn't disable the handling of topology events, so the node list should be up to date.